### PR TITLE
Feature/text search

### PR DESCRIPTION
**Pull request by CorreyL**
## Preamble

The number of commits and number of lines changed started to get a little long for a single PR (imo), so I decided to break it off here, with the caveat that I acknowledge that there's still more features to add based on the initial specifications.

## Still Todo In A Future PR

  * Style `Prev` and `Next` icons when `<button>` is disabled
  * Setting the `textSearchInit` `mode` via the interface
  * A component that'll be rendered as a container with a list of search results

## Known Bug

The first and last `result` cannot be set as an `activeResult` when iterating through the results with the `Prev` and `Next` buttons.

search-prev-next

Figured I'd open this PR as a point of discussion on why this is happening, and how to fix this.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -31,9 +40,19 @@ const App = () => {
...
       docViewer.setToolMode(docViewer.getTool('AnnotationEdit'));
+      setAnnotManager(docViewer.getAnnotationManager());
     });
   }, []);
 
+  /**
+   * Coupled with the function `changeActiveSearchResult`
+   */
+  useEffect(() => {
+    if (activeResultIndex && activeResultIndex >= 0) {
```
Comment (andreysaf) [inline src/App.js]: 0 && true evaluates to 0, which then skips it.

File: src/App.js [hunk 2]
```diff
@@ -61,9 +80,131 @@ const App = () => {
...
+     * Do not try to set a search result that is outside of the index range of
+     * searchResults
+     */
+    if (newSearchResult >= 0 && newSearchResult < searchResults.length) {
+      setActiveResultIndex(newSearchResult);
+    }
+  };
+
+  const prevSearchResult = () => {
+    changeActiveSearchResult(activeResultIndex - 1);
```
Comment (andreysaf) [inline src/App.js]: Make sure you cannot go past 0

File: src/App.js [hunk 3]
```diff
@@ -61,9 +80,131 @@ const App = () => {
...
+  }
+
+  const nextSearchResult = () => {
+    changeActiveSearchResult(activeResultIndex + 1);
```
Comment (andreysaf) [inline src/App.js]: Make sure you cannot go past length of array

**Review by andreysaf — CHANGES_REQUESTED** [review]
Sweet! Looks great and exactly what we are after. I think I found the issue with not being able to jump to the first result.

File: src/App.js (hunk 2 above)
Comment (CorreyL) [reply src/App.js]: Ahh yeah, that was my intention with the `useEffect` guard clause

`if (activeResultIndex && activeResultIndex >= 0)`

But it looks like you took care of it for me in de33243, and I believe things are operating as expected now =D

File: src/App.js (hunk 3 above)
Comment (CorreyL) [reply src/App.js]: Addressed in 991299e

File: src/App.js [hunk 4]
```diff
@@ -61,9 +80,133 @@ const App = () => {
...
+  };
+
+  const prevSearchResult = () => {
+    if (activeResultIndex !== 0) {
+      changeActiveSearchResult(activeResultIndex - 1);
+    }
+  }
+
+  const nextSearchResult = () => {
+    changeActiveSearchResult(activeResultIndex + 1);
```
Comment (andreysaf) [inline src/App.js]: Optionally: do you think it would be possible to update state inside of click handlers for code simplicity?

Comment (CorreyL) [reply src/App.js]: See: 5e8280a

Feel free to let me know if I may have misinterpreted your suggestion!

**Review by andreysaf — APPROVED** [review]

**Pull request merged**